### ISSUE TITLE
[BUG] Upgrading Longhorn-crud to 1.3.0 results in Rancher UI not working

### ISSUE THREAD

**Issue by voarsh2** [longhorn/longhorn#4749]
## Describe the bug
I'm not sure how to explain this, but as soon as I upgraded the CRD to 1.3.0 from 1.2.5 Rancher won't load the UI.
I get this:

![image](https://user-images.githubusercontent.com/60437695/196906662-5084b4bb-695b-4ade-8969-b253af623d91.png)

```
time="2022-10-20T09:04:59Z" level=error msg="Failed to connect to proxy. Empty dialer response" error="dial tcp 192.168.100.97:443: connect: connection refused"
time="2022-10-20T09:04:59Z" level=error msg="Remotedialer proxy error" error="dial tcp 192.168.100.97:443: connect: connection refused"
time="2022-10-20T09:05:09Z" level=info msg="Connecting to wss://192.168.100.97/v3/connect with token starting with 5qp6bxkd96xm67jpxkzppfrdhsf"
time="2022-10-20T09:05:09Z" level=info msg="Connecting to proxy" url="wss://192.168.100.97/v3/connect"
time="2022-10-20T09:05:09Z" level=info msg="Starting plan monitor, checking every 120 seconds"
```

I wasn't able to roll back to before the upgrade:

```
Error: cannot patch "engines.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "engines.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "replicas.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "replicas.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "settings.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "settings.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "volumes.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "volumes.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "engineimages.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "engineimages.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch
 "nodes.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "nodes.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "instancemanagers.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "instancemanagers.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "sharemanagers.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "sharemanagers.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "backingimages.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "backingimages.longhorn.io" is invalid: status.storedVersions[1]
: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "backingimagemanagers.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "backingimagemanagers.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "backingimagedatasources.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io
 "backingimagedatasources.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "backuptargets.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "backuptargets.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "backupvolumes.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "backupvolumes.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "backups.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "backups.
longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions && cannot patch "recurringjobs.longhorn.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "recurringjobs.longhorn.io" is invalid: status.storedVersions[1]: Invalid value: "v1beta2": must appear in spec.versions
```

Ref: https://github.com/longhorn/longhorn/discussions/4198#discussioncomment-3071658
 See attachments for CRDS:
 ```
 kubectl get crds nodes.longhorn.io -o yaml
kubectl get crds engineimages.longhorn.io -o yaml
kubectl get crds backingimages.longhorn.io -o yaml
kubectl get crds backuptargets.longhorn.io -o yaml
kubectl get crds volumes.longhorn.io -o yaml
```

[backingimages.longhorn.io.txt](https://github.com/longhorn/longhorn/files/9828405/backingimages.longhorn.io.txt)
[engineimages.longhorn.io.txt](https://github.com/longhorn/longhorn/files/9828406/engineimages.longhorn.io.txt)
[volumes.longhorn.io.txt](https://github.com/longhorn/longhorn/files/9828407/volumes.longhorn.io.txt)
[backuptargets.longhorn.io.txt](https://github.com/longhorn/longhorn/files/9828408/backuptargets.longhorn.io.txt)

I can't even do an ETCD restore as the cluster agent can't connect, my only choice (besides a solution from the team here), is to roll back an older version of rancher, which shows cluster snapshots (despite the fact that cluster agent doesn't connect, and pray a restore will work, which is not ideal, since it is from the 8th of October........)

Funnily enough I can still access Lonhorn UI.
I tried to lower replica count and got: 
```
unable to update replica count for volume pvc-ad916f17-caa4-42e1-a7ac-353b6275d533: conversion webhook for longhorn.io/v1beta1, Kind=Volume failed: Post "https://longhorn-conversion-webhook.longhorn-system.svc:9443/v1/webhook/conversion?timeout=30s": service "longhorn-conversion-webhook" not found
```

## To Reproduce

1. Go to Helm Apps in Rancher and upgrade Longhorn CRD to 1.3.0 from 1.2.5

## Expected behavior

An upgrade that doesn't bork my cluster

## Log or Support bundle

If applicable, add the Longhorn managers' log or support bundle when the issue happens. 
You can generate a Support Bundle using the link at the footer of the Longhorn UI.

## Environment

 - Longhorn version: 1.2.5
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: RKE
   - Number of management node in the cluster: 5
   - Number of worker node in the cluster: 5
 - Node config
   - OS type and version: Linux 20
   - CPU per node:
   - Memory per node:
   - Disk type(e.g. SSD/NVMe):
   - Network bandwidth between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal):
 - Number of Longhorn volumes in the cluster: 10

**Comment by derekbit** [comment]
Downgrading Longhorn is not supoorted. Why not upgrade to v1.3.1 and check the root cause of the UI not working issue?

**Comment by PhanLe1010** [comment]
It seems that you upgraded the wrong chart. In Rancher Ui, you will see 2 charts `longhorn` and `longhorn-crd`. You should only upgrade the `longhorn` chart. The `longhorn-crd` chart is a dependent chart, its upgrade will be handle automatically by Rancher UI.

<img width="1277" alt="Screen Shot 2022-10-25 at 2 50 02 PM" src="https://user-images.githubusercontent.com/22139961/197888833-0195d927-96b6-4d50-9155-8d1e3d8757d7.png">

**Comment by PhanLe1010** [comment]
> I can't even do an ETCD restore as the cluster agent can't connect, my only choice

Do you mean that the upgrade break Rancher agent as well as the ETCD?

**Comment by voarsh2** [comment]
> It seems that you upgraded the wrong chart. In Rancher Ui, you will see 2 charts `longhorn` and `longhorn-crd`. You should only upgrade the `longhorn` chart. The `longhorn-crd` chart is a dependent chart, its upgrade will be handled automatically by Rancher UI.

Seems I read the upgrade instructions wrong.
In the past I did the crd first, then the main Longhorn one.


> Do you mean that the longhorn-crd chart upgrade broke Rancher agent as well as the ETCD?
As soon as I upgraded Longhorn CRD, Cluster agent had issues, Rancher web UI would basically just time out trying to show the cluster explorer, or any resource for the cluster, it would even hang NOT browsing the cluster itself.

Luckily I was able to initiate a ETCD restore after a lot of trial and error, and once I restored ETCD I uninstalled Longhorn. I've been burned by it for far too long even before it borked my cluster.
